test/Microsoft.AspNet.Mvc.Core.Test/VirtualFileResultTest.cs

```diff
@@ -241,73 +267,29 @@ public void Constructor_SetsContentTypeAndParameters()
-        [InlineData(".\\FilePathResultTestFile.txt")]
-        // Traverse to the parent directory and back to the file system directory
-        [InlineData("..\\TestFiles/FilePathResultTestFile.txt")]
-        [InlineData("..\\TestFiles\\FilePathResultTestFile.txt")]
-        [InlineData("..\\TestFiles/SubFolder/SubFolderTestFile.txt")]
-        [InlineData("..\\TestFiles\\SubFolder\\SubFolderTestFile.txt")]
-        [InlineData("..\\TestFiles/SubFolder\\SubFolderTestFile.txt")]
-        [InlineData("..\\TestFiles\\SubFolder/SubFolderTestFile.txt")]
-        [InlineData("~/FilePathResultTestFile.txt")]
-        [InlineData("~\\TestFiles\\FilePathResultTestFile.txt")]
-        public async Task ExecuteResultAsync_ThrowsFileNotFound_IfFileProviderCanNotFindTheFile(string path)
+        [Fact]
+        public async Task ExecuteResultAsync_ThrowsFileNotFound_IfFileProviderCanNotFindTheFile()
         {
             // Arrange
-            // Point the IFileProvider root to a different subfolder
-            using (var fileProvider = new PhysicalFileProvider(Path.GetFullPath("./Properties")))
+            var path = "TestPath.txt";
+            var fileInfo = new Mock<IFileInfo>();
+            fileInfo.SetupGet(f => f.Exists).Returns(false);
+            var fileProvider = new Mock<IFileProvider>();
+            fileProvider.Setup(f => f.GetFileInfo(path)).Returns(fileInfo.Object);
+            var filePathResult = new VirtualFileResult(path, "text/plain")
             {
-                var filePathResult = new VirtualFileResult(path, "text/plain")
-                {
-                    FileProvider = fileProvider,
-                };
-
-                var expectedMessage = "Could not find file: " + path;
-                var context = new ActionContext(GetHttpContext(), new RouteData(), new ActionDescriptor());
-
-                // Act
-                var ex = await Assert.ThrowsAsync<FileNotFoundException>(() => filePathResult.ExecuteResultAsync(context));
-
-                // Assert
-                Assert.Equal(expectedMessage, ex.Message);
-                Assert.Equal(path, ex.FileName);
-            }
-        }
+                FileProvider = fileProvider.Object,
+            };
 
-        [Theory]
-        [InlineData("/SubFolder/SubFolderTestFile.txt")]
-        [InlineData("\\SubFolder\\SubFolderTestFile.txt")]
-        [InlineData("/SubFolder\\SubFolderTestFile.txt")]
-        [InlineData("\\SubFolder/SubFolderTestFile.txt")]
-        [InlineData("./SubFolder/SubFolderTestFile.txt")]
-        [InlineData(".\\SubFolder\\SubFolderTestFile.txt")]
-        [InlineData("./SubFolder\\SubFolderTestFile.txt")]
-        [InlineData(".\\SubFolder/SubFolderTestFile.txt")]
-        [InlineData("~/SubFolder/SubFolderTestFile.txt")]
-        [InlineData("~/SubFolder\\SubFolderTestFile.txt")]
-        public void ExecuteResultAsync_ThrowsDirectoryNotFound_IfFileProviderCanNotFindTheDirectory(string path)
```

Comment:
This tested a `Assert.ThrowsAsync` without awaiting it. Awaiting it fails the test since we no longer throw `DirectoryNotFoundException`.